router/core/graphql_prehandler.go

```diff
@@ -167,28 +176,83 @@ func (h *PreHandler) Handler(next http.Handler) http.Handler {
 			metrics.Finish(finalErr, statusCode, writtenBytes)
 		}()
 
+		var body []byte
+		var files []httpclient.File
+		var err error
 		// XXX: This buffer needs to be returned to the pool only
 		// AFTER we're done with body (retrieved from parser.ReadBody())
 		buf := pool.GetBytesBuffer()
 		defer pool.PutBytesBuffer(buf)
+		if r.Header.Get("Content-Type") == "" || r.Header.Get("Content-Type") == "application/json" {
```

Comment:
Have you tested the code with a file bigger than 5MB? In other words will our body limiter prevent this?

Reply:
I've tested it with >1Gb

Reply:
Yes, it will. The user need to configure it with a larger threshold

Reply:
This needs to work independently from operations and file uploads otherwise you will open a vulnerability just by using file uploads

Reply:
Done